File: PC_C_Char/PC4_Integrartodo.cs
double nota = 0;
        double suma = 0;
        double contador = 0;
        double mayor = 0;
        double menor = 0;

        while (nota != -1)
        {
            Console.Write("Ingrese nota (0-20) o -1 para salir: ");
            nota = double.Parse(Console.ReadLine());

            if (nota != -1)
            {
                if (nota >= 0 && nota <= 20)
                {
                    suma = suma + nota;
                    contador = contador + 1;

                    if (contador == 1)
                    {
                        mayor = nota;
                        menor = nota;
                    }
                    else
                    {
                        if (nota > mayor)
                            mayor = nota;

                        if (nota < menor)
                            menor = nota;
                    }

                    if (nota >= 17)
                        Console.WriteLine("Excelente");
                    else if (nota >= 14)
                        Console.WriteLine("Bueno");
                    else if (nota >= 11)
                        Console.WriteLine("Aprobado");
                    else
                        Console.WriteLine("Desaprobado");
                }
                else
                {
                    Console.WriteLine("Nota invalida");
                }
            }
        }

        if (contador > 0)
        {
            double promedio = suma / contador;

            Console.WriteLine("Total: " + contador);
            Console.WriteLine("Promedio: " + promedio);
            Console.WriteLine("Mayor: " + mayor);
            Console.WriteLine("Menor: " + menor);
        }
        else
        {
            Console.WriteLine("No hay datos");
        }
    

File: PC_Python/PC4_Integrartodo.py
nota = 0

suma = 0
contador = 0

mayor = 0
menor = 0

while True:

    nota = float(input("Ingrese nota (0-20) o -1 para salir: "))

    if nota == -1:
        break

    if nota >= 0 and nota <= 20:

        suma = suma + nota
        contador = contador + 1

        if contador == 1:
            mayor = nota
            menor = nota
        else:
            if nota > mayor:
                mayor = nota
            if nota < menor:
                menor = nota

        if nota >= 17:
            print("Excelente")
        elif nota >= 14:
            print("Bueno")
        elif nota >= 11:
            print("Aprobado")
        else:
            print("Desaprobado")

    else:
        print("Nota invalida")

if contador > 0:
    promedio = suma / contador
    print("Total:", contador)
    print("Promedio:", promedio)
    print("Mayor:", mayor)
    print("Menor:", menor)
else:
    print("No hay datos")